src/components/document-metadata-modal.js

```diff
@@ -117,11 +118,12 @@ function DocumentMetadataModal({
   }, [isVisible]);
 
   const handleTagSearch = async typedInTag => {
+    const sanitizedTypedInTag = escapeStringRegexp((typedInTag || '').trim());
```

Comment:
Wouldn't it be better to escape the string in the store (directly when constructing the RegExp)? We do this already in another place. Also, in my opinion, the frontend shouldn't have the knowledge how the DB query is done (I mean using RegExp or not).

Reply:
You have a point, the store is a better, safer place for this correction. Will do!

Reply:
Fixed, placed it in the service, that's where we have all sanitizations. Thanks for the suggestion!